KubeArmor/core/kubeUpdate.go

```diff
@@ -2382,12 +2393,103 @@ func (dm *KubeArmorDaemon) removeBackUpPolicy(name string) {
 // == Default Posture == //
 // ===================== //
 
-func validateDefaultPosture(key string, ns *corev1.Namespace, defaultPosture string) string {
+func (dm *KubeArmorDaemon) updatEndpointsWithCM(cm *corev1.ConfigMap, action string) {
+	dm.EndPointsLock.Lock()
+	defer dm.EndPointsLock.Unlock()
+
+	dm.DefaultPosturesLock.Lock()
+	defer dm.DefaultPosturesLock.Unlock()
+
+	// get all namespaces
+	nsList, err := K8s.K8sClient.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
```

Comment:
we need to exit if err != nil as nsList will be nil.
This will cause a sigfault